starfish/test/image/filter/test_api_contract.py

```diff
@@ -0,0 +1,91 @@
+"""
+This module stores a contract for new Filter algorithms.
+
+Contract
+--------
+
+- constructor accepts 3d and 2d data, where default is 2d
+- values emitted by a filter are floats between 0 and 1 (inclusive)
+- exposes a `run`() method
+- run accepts an in-place parameter which defaults to True
+- run always returns an ImageStack (if in-place, returns a reference to the modified input data)
+- run accepts an `n_processes` parameter which determines
+- run accepts a `verbose` parameter, which triggers tqdm to print progress
+
+To add a new filter, simply add default
+parameters for the constructor (omit is_volume), and it will be tested against the contract by
+way of registration to the FilterAlgorithmBase
+"""
+
+from typing import Dict
+
+import numpy as np
+import pytest
+
+from starfish import ImageStack
+from starfish.image import Filter
+
+methods: Dict = Filter._algorithm_to_class_map()
+
+
+def generate_default_data():
+    data = np.random.rand(2, 2, 2, 40, 50).astype(np.float32)
+    return ImageStack.from_numpy_array(data)
+
+
+@pytest.mark.parametrize('filter_class', methods.values())
+def test_all_methods_adhere_to_contract(filter_class):
+    """Test that all filter algorithms adhere to the filtering contract"""
+
+    default_kwargs = filter_class._DEFAULT_TESTING_PARAMETERS
```

Comment:
necro comment: `FilterAlgorithmBase` should be documented to reflect that this magic classvar is a requirement.